### PR TITLE
[SC-6812] Combine deposit withdraw

### PR DESCRIPTION
**Pull request by paszkowskiDamian**

### CODE REVIEW

File: packages/oasis-actions/src/helpers/calculations/Position.ts [hunk 1]
```diff
@@ -158,6 +162,46 @@ export class Position implements IPosition {
...
+    return this._createTargetPosition(
+      amount,
+      ZERO,
+      this._oraclePriceForCollateralDebtExchangeRate,
+      this.debt.amount,
+      this.collateral.amount,
+    )
+  }
+
+  public repay(amount: BigNumber): IPosition {
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/helpers/calculations/Position.ts]: name it `payback` as we have everywhere

File: packages/oasis-actions/src/operations/aave/borrow.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { ADDRESSES } from '../../helpers/addresses'
+import { OPERATION_NAMES } from '../../helpers/constants'
+import { Address } from '../../strategies/types/IPositionRepository'
+
+export interface BorrowArgs {
+  borrowToken: Address
+  amount: BigNumber
+  account: string
+  user: string
+  unwrap: boolean
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/operations/aave/borrow.ts]: can we name it `isEthToken` because it's not clear for me

**Review by marcinciarka — APPROVED** [review]

File: packages/oasis-actions/src/operations/aave/deposit.ts [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+}
+
+export interface DepositArgs {
+  // - either for a swap where the `entryToken` will be exchanged for the `depositToken`
+  // - or it will be directly deposited in the protocol
+  entryTokenAddress: Address
+  entryTokenIsEth: boolean
+  // - either used for a swap if `entryToken` is swapped for `depositToken`
+  // - or it will be directly deposited in the protocol
+  amount: BigNumber
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/operations/aave/deposit.ts]: change it `amountInBaseUnit`

File: packages/oasis-actions/src/operations/aave/deposit.ts [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+  depositToken: Address
+  // Used to pull tokens from if ERC20 is used in the deposit
+  depositorAddress: Address
+  swapArgs?: SwapArgs
+}
+
+export function getIsSwapNeeded(
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/operations/aave/deposit.ts]: should it be exported?

File: packages/oasis-actions/src/operations/aave/deposit.ts [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+
+  return !(sameTokens || ethToWeth)
+}
+
+function getSwapCalls(
+  depositTokenAddress: Address,
+  entryTokenAddress: Address,
+  amount: BigNumber,
+  swapArgs: SwapArgs | undefined,
+  ETH: string,
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/operations/aave/deposit.ts]: `ETHAddress` and `WETHAddress`

File: packages/oasis-actions/src/strategies/aave/depositBorrow.ts [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  }
+}
+
+export async function depositBorrow(
+  {
+    entryToken,
+    slippage,
+    borrowAmount,
+    collectFeeFrom,
+  }: {
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/strategies/aave/depositBorrow.ts]: make interface from it

File: packages/oasis-actions/src/strategies/aave/depositBorrow.ts [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+  const operation = await operations.aave.depositBorrow(depositArgs, borrowArgs)
+
+  /*
+    Final position calculated using actual swap data and the latest market price
+   */
+  const finalPosition: IPosition = dependencies.currentPosition
+    .deposit(collateralDelta)
+    .borrow(debtDelta)
+
+  console.log(JSON.stringify(operation, null, 4))
```
Comment (jakubswierczek) [inline packages/oasis-actions/src/strategies/aave/depositBorrow.ts]: remove log

File: test/aave/DepositBorrow.test.ts [hunk 8]
```diff
@@ -0,0 +1,334 @@
...
+            from: debtToken.precision,
+            to: collateralToken.precision,
+          }),
+          proxy: proxy,
+          user: userAddress,
+          isDPMProxy: true,
+        },
+      )
+
+      console.log(JSON.stringify(positionTransition), null, 4)
```
Comment (jakubswierczek) [inline test/aave/DepositBorrow.test.ts]: remove logs

**Review by jakubswierczek — APPROVED** [review]

**Pull request merged**